provider/flags/provider_test.go

```diff
@@ -25,6 +25,11 @@ func TestParseFlags(t *testing.T) {
 			in:   []string{"--foo", "--bar", "--buz"},
 			out:  map[string]string{"foo": "true", "bar": "true", "buz": "true"},
 		},
+		{
```

Comment:
could you add test with multiple type of format.

Something like: `[]string{"--foo", "--no-bar", "--buz", "biz", "foo"}`